process/transactionProcessor.go

```diff
@@ -435,31 +424,78 @@ func (tp *TransactionProcessor) computeTransactionStatus(tx *transaction.ApiTran
 	if tx.Status != transaction.TxStatusSuccess {
 		return tx.Status
 	}
-	if findIdentifierInLogs(tx, txFailedEvent) {
+
+	if checkIfMoveBalanceNotarized(tx) {
+		return tx.Status
+	}
+
+	txLogsOnFirstLevel := []*transaction.ApiLogs{tx.Logs}
+	if checkIfFailed(txLogsOnFirstLevel) {
+		return transaction.TxStatusFail
+	}
+
+	allLogs, err := tp.gatherAllLogs(tx)
+	if err != nil {
+		log.Warn("error in TransactionProcessor.computeTransactionStatus", "error", err)
+		return data.TxStatusUnknown
+	}
+	if checkIfFailed(allLogs) {
 		return transaction.TxStatusFail
 	}
-	containsCompletion := findIdentifierInLogs(tx, txCompletedEvent) || findIdentifierInLogs(tx, scDeployEvent)
+
+	containsCompletion := findIdentifierInLogs(allLogs, core.CompletedTxEventIdentifier)
 	if containsCompletion {
 		return tx.Status
 	}
 
+	return transaction.TxStatusPending
+}
+
+func checkIfFailed(logs []*transaction.ApiLogs) bool {
+	if findIdentifierInLogs(logs, core.SignalErrorOperation) {
+		return true
+	}
+	if findIdentifierInLogs(logs, internalVMErrorsEventIdentifier) {
```

Comment:
might switch the 2 checks. based on the test data, internal vm errors event seems to appear before the signal error one. just a slight improvement for early exits